tests/models/abiotic/test_abiotic_tools.py

```diff
@@ -31,24 +82,191 @@ def test_calc_molar_density_air(dummy_data):
 
     from virtual_rainforest.models.abiotic import abiotic_tools
 
+    data = dummy_data
+    layers_function = abiotic_tools.set_layers_function(canopy_layers=3, soil_layers=1)
+
     result = abiotic_tools.calc_molar_density_air(
-        temperature=dummy_data.data["temperature_2m"],
-        atmospheric_pressure=dummy_data.data["atmospheric_pressure"],
+        temperature=data["temperature_2m"],
+        atmospheric_pressure=data["atmospheric_pressure"],
     )
     xr.testing.assert_allclose(
-        result, DataArray([120.618713, 119.436176, 118.276602], dims=["cell_id"])
+        result,
+        DataArray(
+            [[120.618713, 119.436176, 118.276602]],
+            dims=["layers", "cell_id"],
+            coords={
+                "layers": [len(layers_function) - 1],
+                "layers_function": ("layers", ["above"]),
+                "cell_id": [0, 1, 2],
+            },
+            name="molar_density_air",
+        ),
     )
 
 
 def test_calc_specific_heat_air(dummy_data):
-    """Test canclualte specific heat of air."""
+    """Test calculate specific heat of air."""
+
+    from virtual_rainforest.models.abiotic import abiotic_tools
+
+    data = dummy_data
+    layers_function = abiotic_tools.set_layers_function(canopy_layers=3, soil_layers=1)
+
+    result = abiotic_tools.calc_specific_heat_air(temperature=data["temperature_2m"])
+
+    xr.testing.assert_allclose(
+        result,
+        DataArray(
+            [[29.214, 29.202, 29.214]],
+            dims=["layers", "cell_id"],
+            coords={
+                "layers": [len(layers_function) - 1],
+                "layers_function": ("layers", [layers_function[-1]]),
+                "cell_id": [0, 1, 2],
+            },
+            name="specific_heat_air",
+        ),
+    )
+
+
+def test_calculate_latent_heat_vaporisation(dummy_data):
+    """Test.
+
+    TODO check why it does not work with dummy_data.
```

Comment:
Has this TODO been resolved?

Reply:
Oh right I see what is happening now

Reply:
OK - just noting that the fixture is included in the sig but then not used.

Reply:
This is still open as the test does not work with the data object, @davidorme explained well why this is the case below. We will need to find a solution for vertical dimensions.